pkg/cmd/pulumi/deployment.go

```diff
@@ -412,46 +404,781 @@ func newDeploymentSettingsDestroyCmd() *cobra.Command {
+		serviceAccount, err = cmdutil.ReadConsoleWithDefault("GCP service account email address",
+			sd.DeploymentSettings.Operation.OIDC.GCP.ServiceAccount)
+	} else {
+		serviceAccount, err = cmdutil.ReadConsole("GCP service account email address")
+	}
+	if err != nil {
+		return err
+	}
+	if serviceAccount == "" {
+		return errors.New("service account email address is required")
+	}
+
+	sd.DeploymentSettings.Operation.OIDC.GCP.ProjectID = projectID
+	sd.DeploymentSettings.Operation.OIDC.GCP.ProviderID = providerID
+	sd.DeploymentSettings.Operation.OIDC.GCP.WorkloadPoolID = workloadPoolID
+	sd.DeploymentSettings.Operation.OIDC.GCP.ServiceAccount = serviceAccount
+
+	return nil
+}
+
+func configureOidcAzure(sd *workspace.ProjectStackDeployment) error {
+	if sd.DeploymentSettings.Operation == nil {
+		sd.DeploymentSettings.Operation = &apitype.OperationContext{}
+	}
+
+	if sd.DeploymentSettings.Operation.OIDC == nil {
+		sd.DeploymentSettings.Operation.OIDC = &apitype.OperationContextOIDCConfiguration{}
+	}
+
+	if sd.DeploymentSettings.Operation.OIDC.Azure == nil {
+		sd.DeploymentSettings.Operation.OIDC.Azure = &apitype.OperationContextAzureOIDCConfiguration{}
+	}
+
+	var clientID string
+	var tenantID string
+	var subscriptionID string
+	var err error
+
+	if sd.DeploymentSettings.Operation.OIDC.Azure.ClientID != "" {
+		clientID, err = cmdutil.ReadConsoleWithDefault("Azure client id", sd.DeploymentSettings.Operation.OIDC.Azure.ClientID)
+	} else {
+		clientID, err = cmdutil.ReadConsole("Azure client id")
+	}
+	if err != nil {
+		return err
+	}
+	if clientID == "" {
+		return errors.New("Client id is required")
+	}
+
+	if sd.DeploymentSettings.Operation.OIDC.Azure.TenantID != "" {
+		tenantID, err = cmdutil.ReadConsoleWithDefault("Azure tenant id", sd.DeploymentSettings.Operation.OIDC.Azure.TenantID)
+	} else {
+		tenantID, err = cmdutil.ReadConsole("Azure tenant id")
+	}
+	if err != nil {
+		return err
+	}
+	if tenantID == "" {
+		return errors.New("Tenant id is required")
+	}
+
+	if sd.DeploymentSettings.Operation.OIDC.Azure.SubscriptionID != "" {
+		subscriptionID, err = cmdutil.ReadConsoleWithDefault("Azure subscription id",
+			sd.DeploymentSettings.Operation.OIDC.Azure.SubscriptionID)
+	} else {
+		subscriptionID, err = cmdutil.ReadConsole("Azure subscription id")
+	}
+	if err != nil {
+		return err
+	}
+	if subscriptionID == "" {
+		return errors.New("Subscription id is required")
+	}
+
+	sd.DeploymentSettings.Operation.OIDC.Azure.ClientID = clientID
+	sd.DeploymentSettings.Operation.OIDC.Azure.TenantID = tenantID
+	sd.DeploymentSettings.Operation.OIDC.Azure.SubscriptionID = subscriptionID
+
+	return nil
+}
+
+func configureAdvancedSettings(displayOpts *display.Options, sd *workspace.ProjectStackDeployment) error {
+	var options []string
+	if err := survey.AskOne(&survey.MultiSelect{
+		Message: "Advanced settings",
+		Options: []string{
+			"Skip automatic dependency installation step",
+			"Skip intermediate deployments",
```

Comment:
Should extract as consts

Reply:
I thought about it, but wasnt sure as they are used only once and colocated. I will move them

Reply:
Sorry, I should have explained more. I suggested this because the literal string values were compared in some places, so using consts helps avoid errors on future updates. I was still picturing them being colocated in the functions rather than defined at the top, but either way is fine.